stonesoup/models/measurement/clutter.py

```diff
@@ -0,0 +1,272 @@
+        doc="The mean of the distribution for normally-distributed clutter. The "
+        "mean is defined in Cartesian space and must have length 2."
+        "This parameter is only needed when using the normal distribution.")
+    covariance: CovarianceMatrix = Property(
+        default=None,
+        doc="Covariance matrix of the distribution for normally-distributed clutter. "
+        "It must be a 2x2 matrix. This parameter is only needed when using the normal "
+        "distribution.")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Check that the chosen distribution matches the parameters given
+        if self.distribution == 'uniform' and self.state_space is None:
+            raise ValueError("To use the uniform distribution, the state_space parameter must be defined")
+        elif self.distribution == 'normal' and self.mean is None and self.covariance is None:
+            raise ValueError("To use the normal distribution, the mean and covariance parameters must be defined")
+        elif self.distribution not in ['uniform', 'normal']:
+            raise ValueError("The clutter model currently supports only uniform or normal (Gaussian) distributions.")
+
+    def function(self, ground_truths: Set[GroundTruthState], position=None, **kwargs) -> Set[Clutter]:
+        """
+        Use the defined distribution and parameters to create simulated clutter
+        for the current time step. Return this clutter to the calling sensor so
+        that it can be added to the measurements.
+
+        Parameters
+        ----------
+        ground_truths : a set of :class:`~.GroundTruthState`
+            The truth states which exist at this time step.
+        position : a 2x1 :class:`~.StateVector` of Cartesian coordinates
+            The position of the sensor which uses this model.
+        
+        Returns
+        -------
+        : set of :class:`~.Clutter`
+            The simulated clutter.
+        """
+        # Extract the timestamp from the ground_truths. Groundtruth is
+        # necessary to get the proper timestamp. If there is no
+        # groundtruth return a set of no Clutter.
+        if not ground_truths:
+            return set()
+
+        # Generate the clutter for this time step
+        timestamp = list(ground_truths)[0].timestamp
+        clutter = set()
+        for _ in range(np.random.poisson(self.clutter_rate)):
+            if self.distribution == 'uniform':
+                random_vector = self.generate_uniform_vector()
+            else:
+                random_vector = self.generate_normal_vector()
+            
+            # Subtract the mounting sensor's position from the clutter,
+            # then convert to polar coordinates.
+            if position is not None:
+                clutter_vector = random_vector - position[0:2].flatten()
+
+            rho, phi = cart2pol(*clutter_vector)
+            clutter_vector = np.array([Bearing(phi), rho])
```

Comment:
Could the measurement model not be used to do this? I guess possibly not as state space may contain more than position?

Reply:
Hmm I'm not sure. Which part specifically were you thinking could be done by the measurement model? Practically the state space should have the same dimension as the position. But I don't know if that is enforced.

Reply:
So the measurement model already has a functions to go from state space to measurement space, as well as logic to handle position offsets (and rotation, that I just realised is missing here).

I'm just wondering if there is a way to use the measurement model that would then remove the need to carry out coordinate conversion step, and such could make it more flexible on which sensors you could use the clutter model (by sharing the measurement model with the sensor).


Maybe something like this? (not tested):
```suggestion
            state = State([0.0] * self.measurement_model.ndim_state, timestamp=timestamp)
            state.state_vector[self.measurement_model.mapping, 0] += random_vector
            
            clutter_vector = self.measurement_model.function(state)
```

Reply:
Oh yes, I like that. And you're right, it now includes rotation which I hadn't thought of before. This change also means that the function() function is now the same for the `ClutterBearingRange` (2D) as for the `ClutterElevationBearingRange` (3D). I implemented this change.

Reply:
This now means that the only difference between `ClutterBearingRange` and `ClutterElevationBearingRange` is the attribute definitions. Do you think I should make the documentation more generic and then have just one class, maybe simply called `ClutterModel`?

Reply:
Due to the other changes made, there is now just one model that incorporates 2D and 3D. It is called `ClutterModel`.